### ISSUE TITLE
[Bug] IDE Fatal Error

### ISSUE THREAD

**Issue by olivernybroe** [laravel-idea/plugin#133]
I got an error from the plugin, however not sure what caused it.

```
com.intellij.diagnostic.PluginException: Incorrect CachedValue use: same CV with different captured context, this can cause unstable results and invalid PSI access.
Field c in class com.laravel_idea.plugin.routing.helpers.d$b has non-equivalent values:
  {Facade/Ignition/Http/Middleware/IgnitionConfigValueEnabled:enableRunnableSolutions=MethodRouteGroupInfo(namespace=, routeNamePrefix=ignition.), routes/web.php=MethodRouteGroupInfo(namespace=Laravel\Horizon\Http\Controllers, routeNamePrefix=)} (java.util.HashMap) and
  {routes/hook.php=MethodRouteGroupInfo(namespace=App\Http\Controllers, routeNamePrefix=), routes/api.php=MethodRouteGroupInfo(namespace=App\Http\Controllers, routeNamePrefix=), vendor/laravel/telescope/src/Http/routes.php=MethodRouteGroupInfo(namespace=, routeNamePrefix=), Facade/Ignition/Http/Middleware/IgnitionConfigValueEnabled:enableRunnableSolutions=MethodRouteGroupInfo(namespace=, routeNamePrefix=ignition.), routes/web.php=MethodRouteGroupInfo(namespace=App\Http\Controllers, routeNamePrefix=)} (java.util.HashMap)
Either make `equals()` hold for these values, or avoid this dependency, e.g. by extracting CV provider into a static method. [Plugin: com.laravel_idea.plugin]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:323)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:59)
	at com.intellij.diagnostic.PluginException.logPluginError(PluginException.java:76)
	at com.intellij.util.CachedValueStabilityChecker.complain(CachedValueStabilityChecker.java:164)
	at com.intellij.util.CachedValueStabilityChecker.checkFieldEquivalence(CachedValueStabilityChecker.java:131)
	at com.intellij.util.CachedValueStabilityChecker.checkProvidersEquivalent(CachedValueStabilityChecker.java:83)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:71)
	at com.laravel_idea.plugin.routing.helpers.d.a(d.java:48)
	at com.laravel_idea.plugin.routing.helpers.c.a(c.java:17)
	at com.laravel_idea.plugin.routing.c.a.a.a(a.java:77)
	at com.laravel_idea.plugin.routing.c.a.a.a(a.java:74)
	at com.laravel_idea.plugin.routing.c.a.a.c(a.java:27)
	at com.laravel_idea.plugin.routing.c.a.e.resolve(e.java:38)
	at com.laravel_idea.plugin.routing.c.a.a.isReferenceTo(a.java:67)
	at com.intellij.psi.search.SingleTargetRequestResultProcessor.processTextOccurrence(SingleTargetRequestResultProcessor.java:38)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl$5.lambda$execute$0(PsiSearchHelperImpl.java:948)
	at com.intellij.psi.impl.search.LowLevelSearchUtil.processTreeUp(LowLevelSearchUtil.java:87)
	at com.intellij.psi.impl.search.LowLevelSearchUtil.lambda$processElementsAtOffsets$0(LowLevelSearchUtil.java:178)
	at com.intellij.psi.impl.search.LowLevelSearchUtil.processOffsets(LowLevelSearchUtil.java:204)
	at com.intellij.psi.impl.search.LowLevelSearchUtil.processElementsAtOffsets(LowLevelSearchUtil.java:177)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl$5.execute(PsiSearchHelperImpl.java:944)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl$2.processInReadAction(PsiSearchHelperImpl.java:280)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl$2.processInReadAction(PsiSearchHelperImpl.java:271)
	at com.intellij.openapi.application.ReadActionProcessor.lambda$process$0(ReadActionProcessor.java:25)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:886)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.application.ReadActionProcessor.process(ReadActionProcessor.java:25)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processCandidates$18(PsiSearchHelperImpl.java:905)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processVirtualFile$11(PsiSearchHelperImpl.java:543)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1138)
	at com.intellij.openapi.application.ex.ApplicationUtil.tryRunReadAction(ApplicationUtil.java:33)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processVirtualFile(PsiSearchHelperImpl.java:525)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processPsiFileRoots$7(PsiSearchHelperImpl.java:400)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processFilesConcurrentlyDespiteWriteActions$8(PsiSearchHelperImpl.java:473)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:183)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processFilesConcurrentlyDespiteWriteActions$9(PsiSearchHelperImpl.java:472)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:136)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:149)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:104)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:86)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by olivernybroe** [comment]
@adelf Alright so I just got this exception again.
This time running v3.4.2.203
```
com.intellij.diagnostic.PluginException: Incorrect CachedValue use: same CV with different captured context, this can cause unstable results and invalid PSI access.
Field b in class com.laravel_idea.plugin.routing.helpers.b$a has non-equivalent values:
  {} (java.util.HashMap) and
  {routes/api.php=MethodRouteGroupInfo(namespace=App\Http\Controllers, routeNamePrefix=), vendor/laravel/telescope/src/Http/routes.php=MethodRouteGroupInfo(namespace=, routeNamePrefix=), vendor/laravel/cashier/routes/web.php=MethodRouteGroupInfo(namespace=Laravel\Cashier\Http\Controllers, routeNamePrefix=cashier.), vendor/laravel/horizon/routes/web.php=MethodRouteGroupInfo(namespace=Laravel\Horizon\Http\Controllers, routeNamePrefix=), Facade/Ignition/Http/Middleware/IgnitionConfigValueEnabled:enableRunnableSolutions=MethodRouteGroupInfo(namespace=, routeNamePrefix=ignition.), routes/web.php=MethodRouteGroupInfo(namespace=App\Http\Controllers, routeNamePrefix=), routes/hook.php=MethodRouteGroupInfo(namespace=App\Http\Controllers, routeNamePrefix=)} (java.util.HashMap)
Either make `equals()` hold for these values, or avoid this dependency, e.g. by extracting CV provider into a static method. [Plugin: com.laravel_idea.plugin]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:323)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:59)
	at com.intellij.diagnostic.PluginException.logPluginError(PluginException.java:76)
	at com.intellij.util.CachedValueStabilityChecker.complain(CachedValueStabilityChecker.java:164)
	at com.intellij.util.CachedValueStabilityChecker.checkFieldEquivalence(CachedValueStabilityChecker.java:131)
	at com.intellij.util.CachedValueStabilityChecker.checkProvidersEquivalent(CachedValueStabilityChecker.java:83)
	at com.intellij.util.CachedValuesManagerImpl.getCachedValue(CachedValuesManagerImpl.java:71)
	at com.laravel_idea.plugin.routing.helpers.b.b(b.java:47)
	at com.laravel_idea.plugin.routing.helpers.g.a(g.java:17)
	at com.laravel_idea.plugin.routing.c.a.a.a(a.java:77)
	at com.laravel_idea.plugin.routing.c.a.a.a(a.java:74)
	at com.laravel_idea.plugin.routing.c.a.a.c(a.java:27)
	at com.laravel_idea.plugin.routing.c.a.e.resolve(e.java:38)
	at com.laravel_idea.plugin.routing.c.a.a.isReferenceTo(a.java:67)
	at com.intellij.psi.search.SingleTargetRequestResultProcessor.processTextOccurrence(SingleTargetRequestResultProcessor.java:38)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl$5.lambda$execute$0(PsiSearchHelperImpl.java:948)
	at com.intellij.psi.impl.search.LowLevelSearchUtil.processTreeUp(LowLevelSearchUtil.java:87)
	at com.intellij.psi.impl.search.LowLevelSearchUtil.lambda$processElementsAtOffsets$0(LowLevelSearchUtil.java:178)
	at com.intellij.psi.impl.search.LowLevelSearchUtil.processOffsets(LowLevelSearchUtil.java:204)
	at com.intellij.psi.impl.search.LowLevelSearchUtil.processElementsAtOffsets(LowLevelSearchUtil.java:177)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl$5.execute(PsiSearchHelperImpl.java:944)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl$2.processInReadAction(PsiSearchHelperImpl.java:280)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl$2.processInReadAction(PsiSearchHelperImpl.java:271)
	at com.intellij.openapi.application.ReadActionProcessor.lambda$process$0(ReadActionProcessor.java:25)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:886)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.application.ReadActionProcessor.process(ReadActionProcessor.java:25)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processCandidates$18(PsiSearchHelperImpl.java:905)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processVirtualFile$11(PsiSearchHelperImpl.java:543)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1133)
	at com.intellij.openapi.application.ex.ApplicationUtil.tryRunReadAction(ApplicationUtil.java:33)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processVirtualFile(PsiSearchHelperImpl.java:525)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processPsiFileRoots$7(PsiSearchHelperImpl.java:400)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processFilesConcurrentlyDespiteWriteActions$8(PsiSearchHelperImpl.java:473)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:183)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.lambda$processFilesConcurrentlyDespiteWriteActions$9(PsiSearchHelperImpl.java:472)
	at com.intellij.concurrency.JobLauncherImpl.lambda$processImmediatelyIfTooFew$2(JobLauncherImpl.java:142)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.concurrency.JobLauncherImpl.lambda$processImmediatelyIfTooFew$3(JobLauncherImpl.java:138)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:856)
	at com.intellij.concurrency.JobLauncherImpl.processImmediatelyIfTooFew(JobLauncherImpl.java:149)
	at com.intellij.concurrency.JobLauncherImpl.invokeConcurrentlyUnderProgress(JobLauncherImpl.java:45)
	at com.intellij.concurrency.JobLauncher.invokeConcurrentlyUnderProgress(JobLauncher.java:49)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processFilesConcurrentlyDespiteWriteActions(PsiSearchHelperImpl.java:489)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processPsiFileRoots(PsiSearchHelperImpl.java:397)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processCandidates(PsiSearchHelperImpl.java:900)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processGlobalRequests(PsiSearchHelperImpl.java:884)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processGlobalRequestsOptimized(PsiSearchHelperImpl.java:836)
	at com.intellij.psi.impl.search.PsiSearchHelperImpl.processRequests(PsiSearchHelperImpl.java:761)
	at com.intellij.psi.search.SearchRequestQuery.processResults(SearchRequestQuery.java:24)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:99)
	at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:116)
	at com.intellij.util.MergeQuery.processResults(MergeQuery.java:22)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:99)
	at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:116)
	at com.intellij.util.UniqueResultsQuery.processResults(UniqueResultsQuery.java:39)
	at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:99)
	at com.intellij.util.AbstractQuery.forEach(AbstractQuery.java:91)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedSymbolsCache.findAnyReference(PhpUnusedSymbolsCache.java:66)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedGlobalSymbolsPass$UnusedElementsData.searchAndCacheReferences(PhpUnusedGlobalSymbolsPass.java:301)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedGlobalSymbolsPass$UnusedElementsData.searchInCandidateFiles(PhpUnusedGlobalSymbolsPass.java:293)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedGlobalSymbolsPass.searchForUnusedSymbols(PhpUnusedGlobalSymbolsPass.java:147)
	at com.jetbrains.php.lang.inspections.deadcode.local.PhpUnusedGlobalSymbolsPass.doCollectInformation(PhpUnusedGlobalSymbolsPass.java:80)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:401)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1138)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:394)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:393)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:369)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:183)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:367)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:188)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by olivernybroe** [comment]
I think this exception came while I opened a secondary project while having another project open at the same time, not completely sure though

**Comment by adelf** [comment]
Fixed in 3.5
